File: backend/src/main/java/com/example/backend/model/VehicleDeploymentService.java
package com.example.backend.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "VehicleDeploymentService")
public class VehicleDeploymentService {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String serviceName;

    private String description;

    @OneToMany(mappedBy = "vehicleDeploymentService", cascade = CascadeType.ALL)
    private List<VehicleDeploymentPlanning> vehicleDeploymentPlannings;
}

File: frontend/src/app/vehicle-deployment-service/VehicleDeploymentService.ts
import { VehicleDeploymentPlanning } from '../vehicle-deployment-planning/VehicleDeploymentPlanning';

export interface VehicleDeploymentService {
  id: number | null;
  serviceName: string;
  description: string;
  vehicleDeploymentPlannings: VehicleDeploymentPlanning[];
}
